src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -857,27 +860,161 @@ private void InitializeComponentsUsingEnvironment(DocumentDatabase documentDatab
             _indexStorage.Initialize(documentDatabase, environment);
 
             IndexPersistence?.Dispose();
+
             SearchEngineType = IndexStorage.ReadSearchEngineType(Name, environment);
+            IndexFieldsPersistence = new IndexFieldsPersistence(this);
+            IndexFieldsPersistence.Initialize();
 
             switch (SearchEngineType)
             {
                 case SearchEngineType.None:
                 case SearchEngineType.Lucene:
-                    IndexPersistence = new LuceneIndexPersistence(this, documentDatabase.IndexStore.IndexReadOperationFactory);
                     SearchEngineType = SearchEngineType.Lucene;
+                    IndexPersistence = new LuceneIndexPersistence(this, documentDatabase.IndexStore.IndexReadOperationFactory);
+
                     break;
                 case SearchEngineType.Corax:
-                    IndexPersistence = new CoraxIndexPersistence(this, documentDatabase.IndexStore.IndexReadOperationFactory);
                     SearchEngineType = SearchEngineType.Corax;
+                    IndexPersistence = new CoraxIndexPersistence(this, documentDatabase.IndexStore.IndexReadOperationFactory);
                     break;
                 default:
                     throw new InvalidDataException($"Cannot read search engine type for {Name}. Please reset the index.");
             }
 
             IndexPersistence.Initialize(environment);
+            EnsureIsInitialized();
+        }
 
-            IndexFieldsPersistence = new IndexFieldsPersistence(this);
-            IndexFieldsPersistence.Initialize();
+        private void EnsureIsInitialized()
```

Comment:
Cannot we do it inside `IndexPersistence.Initialize(environment);`? Then we can add the dictionary creation code selectively for Corax so we won't have to do any checks like:

```
if (SearchEngineType == SearchEngineType.Lucene)
                return;
```